iceoryx_hoofs/test/moduletests/test_posix_mutex.cpp

```diff
@@ -134,4 +134,39 @@ TEST_F(Mutex_test, TryLockWithRecursiveMutexReturnsFalseWhenMutexLockedInOtherTh
     sutRecursive.unlock();
     lockThread.join();
 }
+
+TEST_F(Mutex_test, LockedMutexBlocks)
+{
+    std::atomic_bool isLockFinished{false};
+    sutNonRecursive.lock();
+
+    std::thread lockThread([&] {
+        sutNonRecursive.lock();
+        isLockFinished.store(true);
+        sutNonRecursive.unlock();
+    });
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Please be more creative than just adding big sleeps. Our CI is already slow.

A possible solution would be to
- lock the mutex in the main thread
- start a thread and signal the main thread to be ready
  - get the current time
  - get the lock
  - measure the time it takes to get the lock
  - check if the duration is larger than the sleep time in the main thread
- in the main thread wait for the ready signal from the lockThread
- sleep a few milliseconds (5 to 10)
- release the lock
- join the threads